modules/db/src/main/java/org/mule/module/db/internal/domain/connection/DefaultDbConnection.java

```diff
@@ -393,22 +404,52 @@ public int getNetworkTimeout() throws SQLException
     @Override
     protected void resolveLobs(String typeName, Object[] attributes) throws SQLException
     {
-        try (ResultSet resultSet = this.getMetaData().getAttributes(this.getCatalog(), null, typeName, null))
+        Map<Integer, List> dataTypes = getDataTypes(typeName);
+
+        for(Integer key : dataTypes.keySet())
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Iterate over the "entrySet" instead of the "keySet". ([squid:S2864](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2864))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7681&issues=AWjt-QEVotIXnmsUG4vb&open=AWjt-QEVotIXnmsUG4vb)

[ISSUE_KEY]: AWjt-QEVotIXnmsUG4vb